src/include/common/vector/value_vector.h

```diff
@@ -146,61 +146,54 @@ struct KUZU_API BlobVector {
     }
 };
 
-// Currently, ListVector is used for both VAR_LIST and ARRAY physical type
+// ListVector is used for both LIST and ARRAY physical type
 class KUZU_API ListVector {
 public:
+    static const ListAuxiliaryBuffer& getBuffer(const ValueVector& vector) {
+        return vector.auxiliaryBuffer->constCast<ListAuxiliaryBuffer>();
+    }
+    static ListAuxiliaryBuffer& getBufferUnsafe(const ValueVector& vector) {
+        return vector.auxiliaryBuffer->cast<ListAuxiliaryBuffer>();
+    }
+    // If you call setDataVector during initialize, there must be a followed up
+    // copyListEntryAndBufferMetaData at runtime.
+    // TODO(Xiyang): try to merge setDataVector & copyListEntryAndBufferMetaData
```

Comment:
Can we fix this TODO in this PR? I feel it weird to have separate `setDataVector` and `copyListEntryAndBufferMetaData`.

Reply:
I'm leaving this TODO because I cannot fix it now. For `properties()` the idx of data vector is only available at compile time. I cannot do it at runtime.